pkg/network/server_test.go

```diff
@@ -384,7 +385,8 @@ func startTestServer(t *testing.T, protocolCfg ...func(*config.Blockchain)) *Ser
 	} else {
 		s = newTestServer(t, srvCfg)
 	}
-	startWithCleanup(t, s)
+	go s.Start()
+	defer s.Shutdown()
```

Comment:
Not suitable here. Read how `defer` is executed: https://go.dev/tour/flowcontrol/12.

`s.Shutdown()` need to be called directly in the test code.

Reply:
so tried to use `s.Shutdown()` inside tests, then tried to move `go s.Start()` with `s.Shutdown()` to the tests. Both ways got data race in network tests.

Reply:
`s.Shutdown()` does not wait for all goroutines of the node server to stop, which causes logs to attempt to write after the test has ended. One way to solve it is using `require.Eventually` in tests, as I did in the last commit, another way is to use WaitGroup inside the Shutdown function (which seems to me not a good decision(it is probably not for nothing that it hasn't been used before)).  
@AnnaShaleva @roman-khimov Are there any other possible solutions to this problem?

Reply:
> s.Shutdown() does not wait for all goroutines of the node server to stop,

Well, this is the real source of the problem, we need to wait before the exit. Take a look at the other services and how they wait for all routines, for example at Blockchain or RPC server.